### DISCUSSION TITLE
How to pass refs to lazy load components

### DISCUSSION THREAD

**Discussion by junaid52** [vercel/next.js#69182] [Help] [answered]
### Summary

Hello guys I'm trying to pass refs to lazy load component but getting null instead of {current: null} please check where I'm doing mistake.
`'use client';
import dynamic from 'next/dynamic';
import { useRef } from 'react';
const Component = dynamic(() => import('../components/Component'), {
  ssr: false,
});

export default function Home() {
  const ref = useRef(null);
  return <Component ref={ref} />;
}`
`import { forwardRef, useEffect } from 'react';
const Component = forwardRef((props, ref) => {
  useEffect(() => {
    console.log(ref);
  }, []);
  return <h1>Practice</h1>;
});
export default Component; `

**Comment by icyJoseph** [comment]
The code formatting is very bad, so let's see...

Let's assume you have:

```tsx
'use client';
import dynamic from 'next/dynamic';
import { useRef } from 'react';

const Component = dynamic(() => import('./Component'), {
  ssr: false,
});

export default function Home() {
  const ref = useRef(null);

  return <Component ref={ref} />;
}

```

And then Component is:

```tsx
import { forwardRef, useEffect } from 'react';

const Component = forwardRef((props, ref) => {
  useEffect(() => {
    console.log(ref);
  }, []);

  return <p>Practice</p>;
});

export default Component;
```

Then in this case, it does make sense that the ref is `null`, because it is never attached to a DOM element, nor do you ever use it to hold anything really. 

Could you describe more of what you want to do? With proper code fencing as well?

**Reply by junaid52** [reply]
I attached the ref with p element but getting null in useEffect

**Reply by icyJoseph** [reply]
Ok, with all assumptions out of the way, the way to fix this is by doing:

```tsx
import { forwardRef, useEffect } from 'react';

const Component = (props) => {
  useEffect(() => {
    console.log(props.extRef);
  }, []);

  return <p ref={props.extRef}>Practice</p>;
}

export default Component;

```

And

```tsx
'use client';
import dynamic from 'next/dynamic';
import { useRef } from 'react';

const Component = dynamic(() => import('./Component'), {
  ssr: false,
});

export default function Home() {
  const ref = useRef(null);

  return <Component extRef={ref} />;
}

```

**Reply by junaid52** [reply]
Why there is need of forward ref?

**Reply by icyJoseph** [reply, marked as answer]
> Why there is need of forward ref?

Ah right, I forgot to remove it.

Forward ref exists because, the literal "ref", prop is not passed to function components. So you had to tell React to carry it. However, any other prop name works... It's just that, the ref name, is a convention that's helpful. Now when React 19 comes out, forwardRef won't be a thing anymore

**Reply by junaid52** [reply]
Thankyou so much for your useful answer
